steveify1/multi-tenant-blog-api
# #2 feature(auth):implement signup feature

#### What does this PR do?
- implement and test the sign up feature
#### Description of Task to be completed?
- implement the `POST /auth/signup` endpoint
- document the endpoint with Postman
- implement the `authController` and `AuthenticationService` to manage user authentication
- write tests for the implementation of the signup feature
#### How should this be manually tested?
* clone the repository
* cd into the project directory
* run `npm install`
* run `npm test` to run all the tests (not required)
* run `npm run dev` to start the development server
* send a `POST` request to the `/auth/signup` route. The request body should be in the following format
```
{
 "first_name": <string>,
 "last_name": <string>,
 "email_name": <string>,
 "password": <string>,
 "confirm_password": <string>
}
```
#### Any background context you want to provide?
* The base URL of the API is `http://localhost:5000/api/v1`
* Two Postgresql databases, `mblog` and `mblog_test`, must be created to start the server successfully
* The server will return a success response if the email is not taken, passwords match, and there is no internal error.
#### What are the relevant pivotal tracker stories?
#2 #5 
#### Screenshots (if appropriate)
#### Questions:
